### PR TITLE
Add ShaderValidationTest fixture

### PR DESCRIPTION
**Pull request by kainino0x**
And basic tests to make sure it's working.

-----

- [x] New helpers, if any, are documented in `helper_index.md`.
- [x] Incomplete tests, if any, are marked with TODO or `.unimplemented()`.

**[Review requirements](https://github.com/gpuweb/cts/blob/main/docs/reviews.md) are satisfied for:**

- [x] WebGPU readability
- [x] TypeScript readability

### CODE REVIEW

File: src/webgpu/shader/validation/shader_validation_test.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+   *
+   * @example
+   * ```ts
+   * t.expectCompileResult(true, `wgsl code`); // Expect success
+   * t.expectCompileResult(false, `wgsl code`); // Expect validation error with any error string
+   * t.expectCompileResult('v-0000', `wgsl code`); // Expect validation error containing 'v-0000'
+   * ```
+   */
+  expectCompileResult(result: boolean | string, code: string) {
+    // If no result is expected, we let the scope surrounding the test catch it.
```
Comment (austinEng) [inline src/webgpu/shader/validation/shader_validation_test.ts:15]: nit: perhaps this comment should be reversed: "if an error is expected, push a scope.."

**Review by austinEng — APPROVED** [review]
Does this rely on Chrome-specific behavior? i.e. I think the error messages are supposed to be surfaced through `GPUCompilationInfo` but we don't implement that. Or do the messages go to both places?

**Comment by kainino0x** [comment]
> Does this rely on Chrome-specific behavior? i.e. I think the error messages are supposed to be surfaced through `GPUCompilationInfo` but we don't implement that. Or do the messages go to both places?

Ohh that's a good point, I totally forgot about that. Yes, that makes this chrome-specific behavior; the spec does not require the error message to appear in the validation error message.

**Comment by kainino0x** [comment]
@austinEng WDYT of this approach?

**Comment by austinEng** [comment]
Yup, still looks good

**Pull request merged**